ngraph/frontend/onnx_editor/include/onnx_editor/edge_mapper.hpp

```diff
@@ -0,0 +1,95 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "onnx_editor/editor_types.hpp"
+
+namespace ONNX_NAMESPACE
+{
+    // forward declaration to avoid the necessity of include paths setting in components
+    // that don't directly depend on the ONNX library
+    class GraphProto;
+} // namespace ONNX_NAMESPACE
+
+namespace ngraph
+{
+    namespace onnx_editor
+    {
+        /// \brief A class allows to determine InputEdge and OutputEdge by user-friendly onnx names.
+        class EdgeMapper
+        {
+        public:
+            EdgeMapper() = delete;
+
+            /// \brief Creates an edge mapper based on graph_proto state.
+            ///
+            /// \note If state of graph_proto will be changed, the information
+            ///       from edge mapper is outdated.
```

Comment:
Would it be possible to add an `.update()` method which we could call in this situation?

Reply:
We can introduce sth like:
`void update(const ONNX_NAMESPACE::GraphProto& graph_proto);`
to `EdgeMapper`, but I am not sure if it is useful (especially because OnnxEditor doesn't expose `GraphProto` in public api - I mean after implementation task 51650)

Reply:
OK, I see. So it would be better to recreate the edge mapper after any changes. Makes sense.